builder/dockerfile/shell_parser_test.go

```diff
@@ -33,21 +36,26 @@ func TestShellParser4EnvVars(t *testing.T) {
 		}
 
 		words := strings.Split(line, "|")
-		if len(words) != 2 {
+		if len(words) != 3 {
 			t.Fatalf("Error in '%s' - should be exactly one | in:%q", fn, line)
 		}
 
 		words[0] = strings.TrimSpace(words[0])
 		words[1] = strings.TrimSpace(words[1])
+		words[2] = strings.TrimSpace(words[2])
 
-		newWord, err := ProcessWord(words[0], envs, '\\')
+		// Key W=Windows; A=All; U=Unix
```

Comment:
Can we add a check to make sure that words[0] is W, A or U and not something else?